### PR TITLE
Add a default logo for packs without an image

### PR DESCRIPTION
**Pull request by LizBaker**
## Description

This adds an image component which will conditionally render a supplied logo image or a placeholder.

I took some creative liberty and referenced the stripes from the site logo but am open to any design!

TODO:
- [ ] confirm icon or pack acronym style

## Related Issue(s) / Ticket(s)

Closes https://github.com/newrelic/developer-website/issues/1428

## with an icon

Screen Shot 2021-07-07 at 2 57 16 PM
Screen Shot 2021-07-07 at 2 57 26 PM

## with pack name acronym

Screen Shot 2021-07-08 at 12 42 21 PM
Screen Shot 2021-07-08 at 12 42 37 PM

### CODE REVIEW

**Review by zstix — COMMENTED** [review]
This looks great! I really like the version with the letter, as it gives _some_ uniqueness (rather than the same logo).

One thing we could consider in the future is to add the first letter of each _word_. For example, `AE` for AWS EC2 and `AS` for AWS S3. Probably not something we need to tackle now.

I just had one question about the font resource.

**Review by aswanson-nr — DISMISSED** [review]
Just one small comment! Feel free to disregard it

File: plugins/gatsby-source-newrelic-sdk/gatsby-ssr.js [hunk 1]
```diff
@@ -6,6 +6,11 @@ export const onPreRenderHTML = ({
 }) => {
   replaceHeadComponents([
     ...getHeadComponents(),
+    <link
+      key="space-mono"
+      rel="stylesheet"
+      href="https://fonts.googleapis.com/css2?family=Space+Mono&display=swap"
+    />,
```
[the remark this answers is not in the document]
Comment (LizBaker) [reply plugins/gatsby-source-newrelic-sdk/gatsby-ssr.js:13]: that sounds like a plan! I agree it's not worth it to bring a new font in

File: plugins/gatsby-source-newrelic-sdk/gatsby-ssr.js [hunk 2]
```diff
@@ -6,6 +6,11 @@ export const onPreRenderHTML = ({
 }) => {
   replaceHeadComponents([
     ...getHeadComponents(),
+    <link
+      key="space-mono"
+      rel="stylesheet"
+      href="https://fonts.googleapis.com/css2?family=Space+Mono&display=swap"
```
[the remark this answers is not in the document]
Comment (LizBaker) [reply plugins/gatsby-source-newrelic-sdk/gatsby-ssr.js]: we did this in another repo which I was going by but I saw gatsby has a plugin for this as well! we're gonna scrap the font anyway but good point

**Review by aswanson-nr — APPROVED** [review]

File: plugins/gatsby-source-newrelic-sdk/gatsby-ssr.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (zstix) [reply plugins/gatsby-source-newrelic-sdk/gatsby-ssr.js]: @aswanson-nr out of curiosity, why would you lean towards self-hosting the fonts? With Google's CDN, we stand a chance of the user having the font file cached already. By hosting it ourselves, we guarantee that they'll need to download it when viewing the site.

[the remark this answers is not in the document]
Comment (aswanson-nr) [reply plugins/gatsby-source-newrelic-sdk/gatsby-ssr.js]: Disclaimer: This is very much a nitpicky opinion about performance and control over assets.
1. Fonts are packaged with our artifacts on the CDN, so if google fonts goes down for any reason our users will still see the correct font.
2. Performance-wise there is one less render blocking request to be made when a user navigates to the site.

[This article](https://wpspeedmatters.com/self-host-google-fonts/) goes over self hosting vs Google fonts and has some new points I haven't seen before.

[the remark this answers is not in the document]
Comment (zstix) [reply plugins/gatsby-source-newrelic-sdk/gatsby-ssr.js]: Interesting points, at some point I'd like to do my own research on the subject. This site seems to be a vehicle for him to recommend his own WP plugins, but that doesn't mean he's wrong.

**Comment by LizBaker** [comment]
hihi @kukukaka would you take a look at the screenshots and let us know if you prefer the box icon or the pack name acronym for the placeholder? (and also if you're into the little stripes?)

**Comment by rudouglas** [comment]
I really like the box icon, and would lean more toward that from a design perspective (but I also know nothing about design, just ask my apartment). The only issue I see is that it makes some of the real logos look crappy 😹

**Comment by kukukaka** [comment]
Hi!
Sorry for the late reply.  I really like the idea of the letters, but I think it wouldn't work here so well. Adding just the first letter will not help much to quickly identify the items, compared with other products with the same approach, and I also think the font is not working really well at such big sizes.

So I really like the box icon with the stripes you did. It looks excellent. I would suggest going with that approach if possible and if it's not too late :)

Another unrelated comment would be if we can adjust a bit the paddings of all the logos. I'd double the horizontal padding we have right now, so the logos don't get so close to the border; and add at least the same amount to the top and bottom ones to avoid cases like the Gatsby Build or Java on the screenshots you shared where they have too much mass.

Thanks!

**Pull request merged**